src/super_gradients/training/utils/callbacks/callbacks.py

```diff
@@ -856,6 +856,22 @@ def _infer_global_step(self, context: PhaseContext, is_train_loader: bool):
             return total_steps_in_done + train_loader_length + context.batch_idx
 
 
+@register_callback(Callbacks.SLIDING_WINDOW_INFERENCE)
+class ChangeToSWI(Callback):
```

Comment:
Same as above, I would prefer a longer name that would be more explicit
`ChangeToSWI` -> `ChangeToSlidingWindowInference`.

To further improve the name, I think we can drop the `ChangeTo` and just name it `SlidingWindowInference` or `SlidingWindowValidation` (I think I prefer the second)
And finally, this is just a convention we have for callbacks, let's add `Callback` at the end of the class name.
`SlidingWindowInferenceCallback` or `SlidingWindowValidationCallback`

These are just ideas, feel free to share if you think you have a better name

Reply:
A callback should check whether model supports this at the start of training and raise an exception if model is not compatible.

Reply:
Currently it was implemented for segformer model only.
Fixed on c79d0f2.

Reply:
Yes, I know it is intended only for segformer. However, if someone (a random user) switch the model from segformer to something else and forget to remove this callback from the config, he would get a quite uninformative AttributeError at the call `unwrap_model(context.net).enable_sliding_window_validation()`. It may be obvious to us as code owners what is the problem but not for first-time users.

What I'm suggesting to change here is the following:
```
model =unwrap_model(context.net)
if not hasattr(model, "enable_sliding_window_validation") or not hasattr(model, "disable_sliding_window_validation"):
   raise ConfigurationError("You can sliding windown validation callback, but your model does not support sliding window inference. Please either remove the callback or use the model that supports sliding inference: Segformer, XXX, YYY")
```

That's it. The goal is to just show some love to users and help them understand their mistakes.

Reply:
Fixed on f400bf3

Reply:
``` python
        model = unwrap_model(context.net)
        if not hasattr(model, "enable_sliding_window_validation") or not hasattr(model, "disable_sliding_window_validation"):
            raise ValueError(
                "You can use sliding window validation callback, but your model does not support sliding window "
                "inference. Please either remove the callback or use the model that supports sliding inference: "
                "Segformer"
            )
```
This will be raised even when we dont want to run sliding window. I think this is why your tests failed
Am I right ?

Reply:
Correct. Fixed on ec450ec87e6799ceeb1bc6d25bf3ec75f200a990.

Reply:
Fixed on e2595b598dcb7ace383ba75d31466c470d0e34e7